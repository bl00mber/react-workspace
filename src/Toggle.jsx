class Toggle extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isToggleOn: false };
  }

  handleToggle(e, toggleValue) {
    this.setState(prevState => ({
      isToggleOn: toggleValue !== undefined ? toggleValue : !prevState.isToggleOn
    }));
  }

  render() {
    return (
      <div>
        <p>Is toggled {this.state.isToggleOn ? 'ON' : 'OFF'}</p>
        <button onClick={(e) => this.handleToggle(e)}>Toggle</button>
        <button onClick={(e) => this.handleToggle(e, true)}>Toggle ON</button>
        <button onClick={(e) => this.handleToggle(e, false)}>Toggle OFF</button>
      </div>
    );
  }
}
