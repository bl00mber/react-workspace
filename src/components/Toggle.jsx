import React from 'react';
import PropTypes from 'prop-types';

const Toggle = ({ isToggleOn, onToggle }) => (
  <div>
    <p>
      Is toggled
      {isToggleOn ? 'ON' : 'OFF'}
    </p>
    <button type="button" onClick={() => onToggle()}>Toggle</button>
    <button type="button" onClick={() => onToggle(true)}>Toggle ON</button>
    <button type="button" onClick={() => onToggle(false)}>Toggle OFF</button>
  </div>
);

Toggle.propTypes = {
  isToggleOn: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired,
};

export default Toggle;
