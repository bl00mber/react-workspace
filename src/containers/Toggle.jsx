import { connect } from 'react-redux';

import ToggleComponent from '@/components/Toggle';
import { toggleFlag } from '@/store/actions';

const mapStateToProps = state => ({
  isToggleOn: state.toggle,
});

const mapDispatchToProps = dispatch => ({
  onToggle: flag => dispatch(toggleFlag(flag)),
});

const Toggle = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ToggleComponent);

export default Toggle;
