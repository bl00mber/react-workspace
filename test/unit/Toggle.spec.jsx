import React from 'react';
import renderer from 'react-test-renderer';

import Toggle from '@/Toggle';

describe('Toggle', () => {
  test('should match react-test-renderer snapshot', () => {
    const component = renderer.create(<Toggle />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the toggle
    tree.children[1].props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the toggle
    tree.children[1].props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
