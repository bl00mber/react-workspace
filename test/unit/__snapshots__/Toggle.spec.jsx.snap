// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should match react-test-renderer snapshot 1`] = `
<div>
  <p>
    Is toggled 
    OFF
  </p>
  <button
    onClick={[Function]}
  >
    Toggle
  </button>
  <button
    onClick={[Function]}
  >
    Toggle ON
  </button>
  <button
    onClick={[Function]}
  >
    Toggle OFF
  </button>
</div>
`;

exports[`Toggle should match react-test-renderer snapshot 2`] = `
<div>
  <p>
    Is toggled 
    ON
  </p>
  <button
    onClick={[Function]}
  >
    Toggle
  </button>
  <button
    onClick={[Function]}
  >
    Toggle ON
  </button>
  <button
    onClick={[Function]}
  >
    Toggle OFF
  </button>
</div>
`;

exports[`Toggle should match react-test-renderer snapshot 3`] = `
<div>
  <p>
    Is toggled 
    OFF
  </p>
  <button
    onClick={[Function]}
  >
    Toggle
  </button>
  <button
    onClick={[Function]}
  >
    Toggle ON
  </button>
  <button
    onClick={[Function]}
  >
    Toggle OFF
  </button>
</div>
`;
